### ISSUE TITLE
False-error with function expressions on loadChildren for routes

### ISSUE THREAD

**Issue by Barbiero** [angular/vscode-ng-language-service#341]
Steps to reproduce:

```bash
ng new TodoApp
# Make sure to enable routing
ng g module lazy-loaded --routing=true
```

Then replace the app-routing.module.ts file with this:

```typescript
import { NgModule, Component } from '@angular/core'
import { Routes, RouterModule } from '@angular/router'
import { AppComponent } from './app.component'

@Component({
  template: '<router-outlet></router-outlet>',
})
export class GenericComponent {} // declare GenericComponent on the main module as appropriate

const routes: Routes = [
  {
    path: '',
    component: GenericComponent,
    children: [
      {
        path: './lazyloaded',
        /*
        Error during module compile of 'AppRoutingModule'
          Function expressions are not supported on decorators
            Consider changing the function expression into an exported function
        */
        loadChildren: () => import('./lazy-loaded/lazy-loaded.module').then(m => m.LazyLoadedModule),
      },
    ],
  },
]

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule],
})
export class AppRoutingModule {}
```

The problem is, this error message is a fake: you can run `ng build --aot` and Angular will have no trouble creating this module. In fact, I only noticed this because a recent `ng upgrade` changed the way `loadChildren` were written(previously we used a simple string).

Furthermore, I noticed that if the `GenericComponent` I declared above is not included in this file, the error will go away. It's not uncommon in our project to have a generic "router-outlet" component that would be used as the entrypoint for several child routes, which is why we declare them inside the *-routing modules. Again, the build and execution is working fine, but the language server complains about an error that does not occur.

Angular version: 8.0.1(core & language-service)
VSCode-ng-language-service extension version: 0.800.0

**Comment by Hamcker** [comment]
still seeing this, any solutions?

**Comment by kyliau** [comment]
This has been fixed by the new Ivy-native language service, released in [v11.1.0](https://github.com/angular/vscode-ng-language-service/releases/tag/v11.1.0).
It's an opt-in feature for now, please give it a try and let us know if you have any feedback.
Please make sure your project has [`strictTemplates`](https://angular.io/guide/angular-compiler-options#stricttemplates) enabled in `angularCompilerOptions`.
